### PR TITLE
Depend on apm-data plugin

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Updates docker-compose.yml to enable the apm-data Elasticsearch plugin, which installs the built-in Elasticsearch index templates and ingest pipelines. Update the system tests to stop installing the integration package.

`apm-server.data_streams.wait_for_integration` configuration has been removed. This PR should therefore not be merged until the apm-data Elasticsearch plugin is enabled by default.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~ docs will be updated separately, they're not in this repo anymore

## How to test these changes

Run the system tests.

## Related issues

https://github.com/elastic/apm-server/issues/11528
https://github.com/elastic/apm-server/issues/11529

### CODE REVIEW

**Comment by v1v** [comment]
run docs-build

File: systemtest/approvals/TestRUMErrorSourcemapping/absolute_bundle_filepath/standalone.approved.json [hunk 1]
```diff
@@ -27,192 +30,6 @@
         "error.exception.message": [
             "Uncaught Error: timeout test error"
         ],
-        "error.exception.stacktrace": [
```
Comment (simitt) [inline systemtest/approvals/TestRUMErrorSourcemapping/absolute_bundle_filepath/standalone.approved.json:30]: The mapping changes will need to be documented somewhere; I consider them bugfixes as they result from problems that we have observed with using `flattened` here, but users might have built on top of it already.

**Review by simitt — COMMENTED** [review]
generally approach looks good

File: testing/benchmark/variables.tf [hunk 2]
```diff
@@ -37,6 +37,13 @@ variable "apm_server_zone_count" {
   description = "Optional apm server zone count"
 }
 
+# TODO(axw) disable by default
```
[the remark this answers is not in the document]
Comment (axw) [reply testing/benchmark/variables.tf:40]: This is a leftover, I'm going to revert all these Terraform changes

File: systemtest/approvals/TestRUMErrorSourcemapping/absolute_bundle_filepath/standalone.approved.json (hunk 1 above)
Comment (axw) [reply systemtest/approvals/TestRUMErrorSourcemapping/absolute_bundle_filepath/standalone.approved.json:30]: :+1: I will add a note to the changelog

**Pull request merged**